packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -259,18 +260,49 @@ export class Outbox {
 			true /* disableGroupedBatching */,
 			resubmittingBatchId,
 		);
-		this.flushInternal(
+		this.flushInternalMainBatch(
 			this.mainBatch,
 			false /* disableGroupedBatching */,
 			resubmittingBatchId,
 		);
 	}
 
+	/**
+	 * Flushes the main batch to the ordering service. If the batch is empty, it will create an empty grouped batch
+	 * if BatchId is present (resubmit flow) and the runtime should send.
+	 */
+	private flushInternalMainBatch(
+		mainBatch: BatchManager,
+		disableGroupedBatching: boolean = false,
+		resubmittingBatchId?: BatchId,
+	) {
+		let clientSequenceNumber: number | undefined;
+		if (mainBatch.empty) {
+			if (resubmittingBatchId && this.params.config.immediateMode !== true) {
+				const emptyGroupedBatch = this.params.groupingManager.createEmptyGroupedBatch(
+					resubmittingBatchId,
+					this.params.getCurrentSequenceNumbers().referenceSequenceNumber ?? 0,
```

Comment:
Pretty sure we could assert that `referenceSequenceNumber` is defined.  Let's try it.  If when Loop starts trying it we hit that assert, we can learn from it.

Reasoning being: We're only here if the batch changed on resubmit, and that would only happen if remote changes came in and we had to rebase.  Also, we would have made a connection by this point, so I think we'd have a join op if nothing else.  Feels pretty safe.